### PR TITLE
fix:optimization of QA splitting

### LINKED ISSUE #365: dataprocess pod in example-test failed sometimes
like https://github.com/kubeagi/arcadia/actions/runs/7192102876
```
NAMESPACE            NAME                                                READY   STATUS             RESTARTS        AGE
arcadia              pod/arcadia-apiserver-88c7854b-56v5h                1/1     Running            0               30m
arcadia              pod/arcadia-chromadb-0                              1/1     Running            0               30m
arcadia              pod/arcadia-controller-d8dc76bfc-vks4j              1/1     Running            0               30m
arcadia              pod/arcadia-dataprocess-847886555b-kgz7n            0/1     CrashLoopBackOff   10 (3m7s ago)   30m
arcadia              pod/arcadia-fastchat-6b4b454d69-cqs5g               2/2     Running            0               30m
arcadia              pod/arcadia-minio-57df5cd98c-2b5qq                  1/1     Running            0               30m
arcadia              pod/arcadia-portal-677f6588f4-pvjb7                 1/1     Running            0               30m
kube-system          pod/coredns-57575c5f89-r2w9d                        1/1     Running            0               30m
kube-system          pod/coredns-57575c5f89-xd2n7                        1/1     Running            0               30m
kube-system          pod/etcd-kubeagi-control-plane                      1/1     Running            0               30m
kube-system          pod/kindnet-z2pb9                                   1/1     Running            0               30m
kube-system          pod/kube-apiserver-kubeagi-control-plane            1/1     Running            0               30m
kube-system          pod/kube-controller-manager-kubeagi-control-plane   1/1     Running            0               30m
kube-system          pod/kube-proxy-w95bl                                1/1     Running            0               30m
kube-system          pod/kube-scheduler-kubeagi-control-plane            1/1     Running            0               30m
local-path-storage   pod/local-path-provisioner-c49b7b56f-54dpx          1/1     Running            0               30m
```

pod logs show
```
2023-12-13T08:07:00.668647927Z stderr F Config There is an error when load the config (file path =  config.yml). 
2023-12-13T08:07:00.668681821Z stderr F Traceback (most recent call last):
2023-12-13T08:07:00.668688603Z stderr F   File "/arcadia_app/data_manipulation/common/config.py", line 35, in __init__
2023-12-13T08:07:00.668694024Z stderr F     with open(yaml_file_path, 'r') as file:
2023-12-13T08:07:00.668699955Z stderr F FileNotFoundError: [Errno 2] No such file or directory: 'config.yml'
2023-12-13T08:07:00.668704794Z stderr F 
2023-12-13T08:07:00.668761157Z stderr F Traceback (most recent call last):
2023-12-13T08:07:00.668782207Z stderr F   File "/arcadia_app/data_manipulation/server.py", line 22, in <module>
2023-12-13T08:07:00.668787136Z stderr F     from common.config import config
2023-12-13T08:07:00.668791474Z stderr F   File "/arcadia_app/data_manipulation/common/config.py", line 262, in <module>
2023-12-13T08:07:00.668910752Z stderr F     config = Config()
2023-12-13T08:07:00.668921151Z stderr F   File "/arcadia_app/data_manipulation/utils/class_utils.py", line 28, in __call__
2023-12-13T08:07:00.668990073Z stderr F     cls._instances[cls] = super(Singleton, cls).__call__(*args, **kwargs)
2023-12-13T08:07:00.668999721Z stderr F   File "/arcadia_app/data_manipulation/common/config.py", line 49, in __init__
2023-12-13T08:07:00.669062159Z stderr F     self.__set_property_value(yaml_data)
2023-12-13T08:07:00.669067649Z stderr F   File "/arcadia_app/data_manipulation/common/config.py", line 140, in __set_property_value
2023-12-13T08:07:00.669163219Z stderr F     self.open_ai_default_key = self.__get_value_by_key_in_yaml(
2023-12-13T08:07:00.669171294Z stderr F   File "/arcadia_app/data_manipulation/common/config.py", line 246, in __get_value_by_key_in_yaml
2023-12-13T08:07:00.669270836Z stderr F     value = config_json[parent_key].get(key)
2023-12-13T08:07:00.669280334Z stderr F KeyError: 'open_ai'

```

### PR DESCRIPTION
**Pull request by wangxinbiao**
#### What this PR does / why we need it

fix: optimization of QA splitting
fix: No such file or directory: 'config.yml'

#### Which issue(s) this PR fixes
 Fixes #365

### CODE REVIEW

File: data-processing/data_manipulation/common/config.py [hunk 1]
```diff
@@ -30,6 +31,11 @@ class Config(metaclass=Singleton):
     
     def __init__(self, yaml_file_path='config.yml'):
         logger.debug(f"{log_tag_const.CONFIG} start to load config file.")
+
+        file = Path(yaml_file_path)
+        if not file.exists():
+            yaml_file_path=os.path.join(os.getcwd(), 'data_manipulation/config.yml')
```
Comment (nkwangleiGIT) [inline data-processing/data_manipulation/common/config.py:37]: for config.yaml, if we allow user to select model service, do we still need configuration like zhipuai，llm.use_type and open_ai?

File: data-processing/data_manipulation/common/config.py [hunk 2]
```diff
@@ -70,23 +76,23 @@ def __set_property_value(self, yaml_data):
             parent_key='minio',
             key='access_key',
             env_name='MINIO_ACCESSKEY',
-            default_value='hpU4SCmj5jiU7IP5'
+            default_value='xxxxxx'
```
Comment (nkwangleiGIT) [inline data-processing/data_manipulation/common/config.py:79]: don't use xxxxx as it's meaningless, we can use something like 'faked_minio_ak', 'faked_minio_sk', 'faked_minio_api_url', so we can know what's the wrong value if any issue occurs.

File: data-processing/data_manipulation/config.yml [hunk 3]
```diff
@@ -1,22 +1,21 @@
...
-  qa_retry_count: '${LLM_QA_RETRY_COUNT: 100}'
+  qa_retry_count: '${LLM_QA_RETRY_COUNT: 10}'
 
 open_ai:
   key: '${OPEN_AI_DEFAULT_KEY: fake}'
-  base_url: '${OPEN_AI_DEFAULT_BASE_URL: http://arcadia-fastchat.172.22.96.167.nip.io/v1}'
-  model: '${OPEN_AI_DEFAULT_MODEL_NAME: 3d407a8b-90ab-43c4-9fc0-87e533368570}'
-
+  base_url: '${OPEN_AI_DEFAULT_BASE_URL: xxxxxx}'
+  model: '${OPEN_AI_DEFAULT_MODEL_NAME: xxxxxx}'
```
Comment (nkwangleiGIT) [inline data-processing/data_manipulation/config.yml:18]: if we allow user to select model service, do we still need configuration like zhipuai，llm.use_type and open_ai?

File: data-processing/data_manipulation/common/config.py (hunk 1 above)
Comment (wangxinbiao) [reply data-processing/data_manipulation/common/config.py:37]: don't need

File: data-processing/data_manipulation/config.yml (hunk 3 above)
Comment (wangxinbiao) [reply data-processing/data_manipulation/config.yml:18]: don't need

File: data-processing/data_manipulation/common/config.py (hunk 2 above)
Comment (wangxinbiao) [reply data-processing/data_manipulation/common/config.py:79]: ok

File: data-processing/data_manipulation/llm_prompt_template/open_ai_prompt.py [hunk 4]
```diff
@@ -17,7 +17,7 @@ def get_default_prompt_template():
     prompt_template = """
         {text}
         
-        请将上述内容按照问题、答案成对的方式，提出问题，并给出每个问题的答案，每个问题必须有问题和对应的答案，并严格按照以下方式展示：
+        请将上述内容按照问题、答案成对的方式，最多提出 25 个问题，并给出每个问题的答案，每个问题必须有问题和对应的答案，并严格按照以下方式展示：
```
Comment (bjwswang) [inline data-processing/data_manipulation/llm_prompt_template/open_ai_prompt.py:20]: 问题的数量是否应该可配置？

Comment (nkwangleiGIT) [reply data-processing/data_manipulation/llm_prompt_template/open_ai_prompt.py:20]: 目前 chuck size 500， overlay 50，后续这些应该一起暴露出去允许配置

**Review by bjwswang — CHANGES_REQUESTED** [review]
Please follow this latest dataprocessign configuration to update your code

https://github.com/kubeagi/arcadia/pull/399

File: data-processing/data_manipulation/llm_prompt_template/open_ai_prompt.py (hunk 4 above)
Comment (bjwswang) [reply data-processing/data_manipulation/llm_prompt_template/open_ai_prompt.py]: 可以先设置默认值，然后开放用户页面配置

**Comment by bjwswang** [comment]
@wangxinbiao Since your marked that this pr can fix #365 , please make sure you have tested this locally

File: data-processing/data_manipulation/kube/custom_resources.py [hunk 5]
```diff
@@ -40,6 +40,8 @@ def get_name(self):
 arcadia_resource_datasources = CustomResource(arcadia_group, "datasources")
 # CRD Dataset
 arcadia_resource_datasets = CustomResource(arcadia_group, "datasets")
+# CRD Dataset
```
Comment (nkwangleiGIT) [inline data-processing/data_manipulation/kube/custom_resources.py:43]: CRD LLM ?

File: data-processing/data_manipulation/llm_api_service/qa_provider_open_ai.py [hunk 6]
```diff
@@ -34,21 +37,15 @@ class QAProviderOpenAI(BaseQAProvider):
...
-        if model is None:
-            model = config.open_ai_default_model
-        
-        self.llm = OpenAI(
+        self.llm = ChatOpenAI(
             openai_api_key=api_key, 
             base_url=base_url,
-            model=model
+            model=model,
+            temperature=0.8
```
Comment (nkwangleiGIT) [inline data-processing/data_manipulation/llm_api_service/qa_provider_open_ai.py:48]: temperature and top_p/top_k should be configured later

File: data-processing/data_manipulation/llm_api_service/qa_provider_open_ai.py [hunk 7]
```diff
@@ -68,40 +65,57 @@ def generate_qa_list(
...
+                            f"The tracing error is: \n{traceback.format_exc()}\n"
+                        ]))
+
+                        status = 1000
+                        message = traceback.format_exc()
+
+                        break
+                    else:
+                        logger.warn('failed to get QA list, wait for 5 seconds and retry')
+                        time.sleep(10) # sleep 5 seconds
```
Comment (nkwangleiGIT) [inline data-processing/data_manipulation/llm_api_service/qa_provider_open_ai.py:108]: update comment to 10 seconds or sleep 5 second?

File: data-processing/data_manipulation/llm_api_service/qa_provider_zhi_pu_ai_online.py [hunk 8]
```diff
@@ -73,14 +72,28 @@ def generate_qa_list(
...
+                    'data': result
+                }
         except Exception as ex:
             result = []
             logger.error(''.join([
                 f"{log_tag_const.ZHI_PU_AI} Cannot access the ZhiPuAI service.\n",
                 f"The tracing error is: \n{traceback.format_exc()}\n"
             ]))
+            return {
+                'status': 1001,
```
Comment (nkwangleiGIT) [inline data-processing/data_manipulation/llm_api_service/qa_provider_zhi_pu_ai_online.py:87]: 1001, is this a http code or some customized error code?

File: data-processing/data_manipulation/llm_api_service/qa_provider_zhi_pu_ai_online.py [hunk 9]
```diff
@@ -100,4 +113,52 @@ def __format_response_to_qa_list(self, response):
...
+        ----------
+        text
+            use the text to generate QA list
+        prompt_template
+            the prompt template
+        """
+        try:
+            response = zhipuai.model_api.invoke(
+                model="chatglm_6b",
+                prompt=[{"role": "user", "content": "你好"}],
```
Comment (nkwangleiGIT) [inline data-processing/data_manipulation/llm_api_service/qa_provider_zhi_pu_ai_online.py:132]: 这个 你好的 prompt 是干啥用的？测试吗

**Comment by wangxinbiao** [comment]
> Please follow this latest dataprocessign configuration to update your code
> 
> #399

File: data-processing/data_manipulation/kube/custom_resources.py (hunk 5 above)
Comment (wangxinbiao) [reply data-processing/data_manipulation/kube/custom_resources.py:43]: 获取llm里面的secret信息，然后去secret获取外部的apikey信息

File: data-processing/data_manipulation/llm_api_service/qa_provider_zhi_pu_ai_online.py (hunk 8 above)
Comment (wangxinbiao) [reply data-processing/data_manipulation/llm_api_service/qa_provider_zhi_pu_ai_online.py:87]: customized error code

File: data-processing/data_manipulation/llm_api_service/qa_provider_zhi_pu_ai_online.py (hunk 9 above)
Comment (wangxinbiao) [reply data-processing/data_manipulation/llm_api_service/qa_provider_zhi_pu_ai_online.py:132]: 拿到的apikey有可能是base64加密的，也有可能没加密，这个主要是用来check的，如果抛异常则需要base64解密一下。

File: data-processing/data_manipulation/llm_api_service/qa_provider_open_ai.py (hunk 7 above)
Comment (wangxinbiao) [reply data-processing/data_manipulation/llm_api_service/qa_provider_open_ai.py:108]: sleep 10 second，我改一下日志和注释内容

**Comment by wangxinbiao** [comment]
> @wangxinbiao Since your marked that this pr can fix #365 , please make sure you have tested this locally

tested

**Comment by wangxinbiao** [comment]
> Please follow this latest dataprocessign configuration to update your code
> 
> #399

@bjwswang ok

File: data-processing/data_manipulation/llm_api_service/qa_provider_zhi_pu_ai_online.py (hunk 9 above)
Comment (nkwangleiGIT) [reply data-processing/data_manipulation/llm_api_service/qa_provider_zhi_pu_ai_online.py:132]: 太隐晦了，默认确定好是什么方式加密吧，不要去调用来 check

File: data-processing/data_manipulation/kube/custom_resources.py (hunk 5 above)
Comment (nkwangleiGIT) [reply data-processing/data_manipulation/kube/custom_resources.py:43]: 我是说注释不对

File: data-processing/data_manipulation/llm_api_service/qa_provider_open_ai.py (hunk 6 above)
Comment (nkwangleiGIT) [reply data-processing/data_manipulation/llm_api_service/qa_provider_open_ai.py:48]: add 'TODO: temperature and top_p/top_k should be configured later'

**Review by nkwangleiGIT — APPROVED** [review]
/lgtm
for other TODOs, do it in other PR

**Pull request merged**